src/Microsoft.AspNetCore.Razor/Compilation/TagHelpers/TagHelperRequiredAttributeDescriptor.cs

```diff
@@ -0,0 +1,71 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+
+namespace Microsoft.AspNetCore.Razor.Compilation.TagHelpers
+{
+    public class TagHelperRequiredAttributeDescriptor
+    {
+        public string Name { get; set; }
+
+        public string Value { get; set; }
+
+        public char Selector { get; set; }
+
+        public bool IsCSSSelector { get; set; }
+
+        public bool Matches(string attributeName, string attributeValue)
+        {
+            if (IsCSSSelector)
+            {
+                var nameMatches = string.Equals(Name, attributeName, StringComparison.OrdinalIgnoreCase);
+
+                if (!nameMatches)
+                {
+                    return false;
+                }
+
+                var valueMatches = false;
+
+                switch (Selector)
+                {
+                    case '^': // Value starts with
```

Comment:
Nah, gonna keep as is. Adding an enum for this would be overkill since the knowledge of supported selectors is known only by this class.